src/connector/src/sink/clickhouse.rs

```diff
@@ -91,22 +90,22 @@ impl ClickHouseSink {
 
     /// Check that the column names and types of risingwave and clickhouse are identical
     fn check_column_name_and_type(&self, clickhouse_column: Vec<SystemColumn>) -> Result<()> {
-        let fields = self.schema.fields().clone();
-        assert_eq!(
-            fields.len(),
-            clickhouse_column.len(),
-            "Schema len not match"
-        );
-        fields
+        let ck_fields_name = build_fields_name_type_from_schema(&self.schema)?;
```

Comment:
We are calling `build_fields_name_type_from_schema` for many times. We may consider storing them in a field of sink.